fix(#4970) : font-family

Blakelist7 opened the pull request:
Some of the font-family containing whitespaces were not quoted

<!-- Please provide the following information to help us review this PR: -->


Quoted font-family having whitespaces.

#4970

Blakelist7 committed:
fix: font-family

Some of the font-family containing whitespaces were not quoted

wbamberg commented:
To fix this bug I think the minimum is that anywhere a family name containing spaces is given as example code, it should be quoted.

This means [line 27](https://github.com/mdn/content/blob/ca3b41e998c548e46c17f5918265201ea8343102/files/en-us/web/css/font-family/index.html#L29) and [line 57](https://github.com/mdn/content/blob/ca3b41e998c548e46c17f5918265201ea8343102/files/en-us/web/css/font-family/index.html#L57) (though actually I don't even know why this additional example is included at all, it just duplicates part of the bigger block above).

This does not include places where names are just mentioned in the text, like in lines [69-71](https://github.com/mdn/content/blob/ca3b41e998c548e46c17f5918265201ea8343102/files/en-us/web/css/font-family/index.html#L69-L71), because that's just referencing the font family, it isn't providing example code.

The bit at [lines 108-111](https://github.com/mdn/content/blob/ca3b41e998c548e46c17f5918265201ea8343102/files/en-us/web/css/font-family/index.html#L108-L111) is a special case, because the declaration at line 110 *is* valid, it's just not recommended. So it would be better to clarify the situation in this section, by saying something like:

> Font family names must either be given quoted as strings, or unquoted as a sequence of one or more identifiers. This means that punctuation characters and digits at the start of each token must be escaped in unquoted font family names.
> 
> It is good practice to quote font family names that contain white space, digits, or punctuation characters other than hyphens. 

...then in the examples, we could make the "good" example (coloured green) only include the "font-family: "Goudy Bookletter 1911", sans-serif;" example, and add a third example after the "bad" example, that should be neither green nor red, but precede it like:

> The following example is technically valid but is not recommended:
>
>   ```css
>  font-family: Gill Sans Extrabold, sans-serif;
>   ```

Does that make sense?

Blakelist7 commented:
Yeah I think that do make some sense.

Blakelist7 committed:
fix: quoted font family and added a section

Section was added to let people know why we recommend to quote font family with whitespaces
Example added

Blakelist7 commented:
Required changes done :)

wbamberg commented on files/en-us/web/css/font-family/index.html:
```diff
@@ -54,13 +54,13 @@ <h2 id="Syntax">Syntax</h2>
 <dl>
  <dt><a id="family-name"><code>&lt;family-name&gt;</code></a></dt>
- <dd>The name of a font family. For example, "Times" and "Helvetica" are font families. Font family names containing whitespace should be quoted.</dd>
+ <dd>The name of a font family. For example, "Times" and "Helvetica" are font families. Font family names containing whitespace should be quoted. For example, 'Comic Sans MS'.</dd>
```
```suggestion
 <dd>The name of a font family. For example, "Times" and "Helvetica" are font families. Font family names containing whitespace should be quoted. For example: "Comic Sans MS".</dd>
```

wbamberg reviewed (changes requested):
I had one copy-edit comment, and it looks like we missed https://github.com/mdn/content/blob/f9d710480812705f184665a3bd42de55e9ab098f/files/en-us/web/css/font-family/index.html#L136 and https://github.com/mdn/content/blob/f9d710480812705f184665a3bd42de55e9ab098f/files/en-us/web/css/font-family/index.html#L144 . Sorry!

Blakelist7 committed:
Update files/en-us/web/css/font-family/index.html

Co-authored-by: wbamberg <will@bootbonnet.ca>

Blakelist7 committed:
Update files/en-us/web/css/font-family/index.html

Co-authored-by: Blakelist7 <kanwal.88singh@gmail.com>

Blakelist7 commented:
> I had one copy-edit comment, and it looks like we missed
> 
> https://github.com/mdn/content/blob/f9d710480812705f184665a3bd42de55e9ab098f/files/en-us/web/css/font-family/index.html#L136
> and
> 
> https://github.com/mdn/content/blob/f9d710480812705f184665a3bd42de55e9ab098f/files/en-us/web/css/font-family/index.html#L144
> . Sorry!

No worry. Next time I'll be more careful 😃 
Done with changes :)

Blakelist7 commented:
@wbamberg done with suggestions :D

wbamberg commented on files/en-us/web/css/font-family/index.html:
```diff
@@ -125,7 +125,7 @@ <h3 id="Valid_family_names">Valid family names</h3>
 
 <pre class="brush: css;">
-font-family: Gill Sans Extrabold, sans-serif;
+font-family: "Gill Sans Extrabold", sans-serif;
```
This should *not* be quoted, because it's an example of the not-good practice of not quoting.

Blakelist7 replied:
My bad, did'nt saw that

Blakelist7 committed:
update

Blakelist7 commented:
@wbamberg now ig it's all clear 👀

wbamberg reviewed (approved):
Thanks @Blakelist7 , looks good to me!
